src/auth/index.ts

```diff
@@ -0,0 +1,44 @@
+import { CachedSigner } from "./cached-signer.js";
+import { CachedVerifier } from "./cached-verifier.js";
+import { PayloadSigner } from "./payload-signer.js";
+import { PayloadVerifier } from "./payload-verifier.js";
+
+export const authModes = ["cached", "payload"] as const;
+export type AuthMode = (typeof authModes)[number];
+
+export type Signer = {
+  signature(timestamp: number, body: string): string;
+};
+
+export type SignerOptions = {
+  secretKey: string;
+  expirationTime: number;
+};
+
+export function makeSigner(type: AuthMode, options: SignerOptions): Signer {
+  switch (type) {
+    case "cached":
+      return new CachedSigner(options.secretKey, options.expirationTime);
+    case "payload":
+      return new PayloadSigner(options.secretKey);
```

Comment:
Oh let's not provide both options

Either or, not both

This makes the documentation really confusing because then we need to explain both